cmd/tailscaled/tailscaled.go

```diff
@@ -96,6 +97,34 @@ var subCommands = map[string]*func([]string) error{
 	"debug":                   &debugModeFunc,
 }
 
+func tryWindowsAppDataMigration() {
+	if runtime.GOOS != "windows" {
+		return
+	}
+
+	localAppData := filepath.Join(os.Getenv("LocalAppData"), "Tailscale")
+	_, err := os.Stat(localAppData)
+	if os.IsNotExist(err) {
+		return
+	}
+	if err != nil {
+		log.Printf("tryWindowsAppDataMigration failed; no AppDataLocal: %v", err)
+		return
+	}
+
+	programData := filepath.Join(os.Getenv("ProgramData"), "Tailscale")
+	_, err = os.Stat(programData)
+	if err == nil || os.IsNotExist(err) {
+		err = os.Rename(localAppData, programData)
+		if err != nil {
+			log.Printf("tryWindowsAppDataMigration rename failed: %v", err)
+			return
+		}
+	} else {
+		log.Printf("tryWindowsAppDataMigration ProgramData stat failed: %v", err)
```

Comment:
do this earlier, return early, then outdent the rest?

Reply:
Done.